src/shell_minimal.js

```diff
@@ -43,10 +42,7 @@ var ENVIRONMENT_IS_WEB = !ENVIRONMENT_IS_NODE && !ENVIRONMENT_IS_SHELL;
 var ENVIRONMENT_IS_WEB = !ENVIRONMENT_IS_SHELL;
 #else
 var ENVIRONMENT_IS_WEB = !ENVIRONMENT_IS_NODE;
-#else
-
 #endif
-#endif // ASSERTIONS
```

Comment:
Wouldn't this regress code size? Also, I think minimal runtime might be aiming to not use these variables except as assertions during startup?

Reply:
OK, In that case we need to find and fix the usage of this variable that causes me to notice it was not defined.

Reply:
Actually I'm not sure you are correct.

The previous lines define ENVIRONMENT_IS_NODE and ENVIRONMENT_IS_SHELL when its possible that we are running in those environments: 

```
#if ENVIRONMENT_MAY_BE_NODE                                                                           
var ENVIRONMENT_IS_NODE = typeof process === 'object';                           
#endif                                                                           
                                                                                 
#if ENVIRONMENT_MAY_BE_SHELL                                                     
var ENVIRONMENT_IS_SHELL = typeof read === 'function';                           
#endif    
```

I don't see why ENVIRONMENT_IS_WEB should be special.   Also there are several library functions that depend on this variable existing.

Without this I see `[client logging: /?exception=Uncaught ReferenceError: ENVIRONMENT_IS_WEB is not defined / undefined ]
` in the console.. but actually the tests seem to pass so I can probably revert this chunk from this PR at least.